### PR TITLE
Fixes #230

### LINKED ISSUE #230: Automate generic post script execution
Instead of having users to upload the post script to http server and make it available, we provide a way for users to specify the post script file location in hosts.yaml file.  Snaps-boot will automate the upload and then execute the post script for each machine unless users override it by providing post_script_url themselves.

### PR DESCRIPTION
**Pull request by bo-quan**
#### What does this PR do?
Automate generic post script execution for servers.

#### Do you have any concerns with this PR?
No

#### How can the reviewer verify this PR?
Deploy snaps_boot and verify that the post script is automatically uploaded and executed.

#### Any background context you want to provide?
Currently specifying post script url for each server is sometimes inconvenient. This PR provides a generic post script file upload and execution mechanism.

#### Questions:
- Have you connected this PR to the issue it resolves?
#230 

- Does the documentation need an update?
Yes, in the PR.

- Does this add new Python dependencies?
No

- Have you added unit or functional tests for this PR?
Yes, added a test to validate post script file execution in CI.

- Does this patch update any configuration files?
No

### CODE REVIEW

File: ci/scripts/post_script [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+touch /var/log/hello_world
```
Comment (RandyLevensalor) [inline ci/scripts/post_script]: Shouldn't this end in a new line?

File: ci/snaps/snaps_pxe_tmplt.yaml [hunk 2]
```diff
@@ -482,6 +485,9 @@ ansible:
       snaps_boot_dir:
         type: string
         value: {{ src_copy_dir }}/snaps-boot
+      check_file:
```
Comment (RandyLevensalor) [inline ci/snaps/snaps_pxe_tmplt.yaml]: This only needs to be checked once per server, not on every interface on each server.

File: ci/scripts/post_script (hunk 1 above)
Comment (bo-quan) [reply ci/scripts/post_script]: Good catch. Fixed.

File: ci/snaps/snaps_pxe_tmplt.yaml (hunk 2 above)
Comment (bo-quan) [reply ci/snaps/snaps_pxe_tmplt.yaml]: @RandyLevensalor good point. Fixed in the latest patch.

**Pull request merged**